### ISSUE TITLE
Graphs or statistics!

### ISSUE THREAD

**Issue by JacobEvelyn** [dependabot/feedback#268]
We'd find it **enormously** helpful if Dependabot's web UI could give us some insight into some basic statistics for our apps. We're thinking of things like either a graph or just raw stats on:

- the amount of time Dependabot PRs are left open before merging
    - This would be especially useful if we could see it along any of these dimensions:
        - min. time left open, max. time left open, avg. time left open
        - per-app vs. overall
        - security PRs vs. non-security PRs vs. all PRs
        - overall vs. sliding window of the past ~month
- the number of dependencies that are out-of-date over time (per-app and overall)
    - This would help us see if when we start using Dependabot on a repo with a lot of out-of-date dependencies, are we merging Dependabot PRs faster than new ones are coming in? (This is especially useful for apps that still see this message: `This repo was added to Dependabot recently, so you'll receive a maximum of 5 PRs for your first few update runs.`)

I get that this is probably a big ask, but even just a CSV dump would be hugely helpful for answering the big questions like:

1. Are we "keeping up" with Dependabot, or are we getting further and further behind?
2. How prompt are we with security patches?

These questions aren't necessarily hard to answer with one or two apps using Dependabot, but as we expand to dozens of apps with hundreds or thousands of recursive dependencies each, it gets really hard to know if we're prioritizing our time effectively here.

While we could potentially get this data ourselves via the GitHub API, we'd have to be mindful of lots of edge cases that Dependabot might have more insight into (for example, when an open Dependabot PR is closed because Dependabot opens a newer one that supersedes it, should that be treated as just one "continuous" PR from a stats point of view?).

Thoughts?

**Comment by mizhi** [comment]
I would personally love to simply have a statistic of how many PRs are opened each week.

Or, given a project's dependencies, what is the rate of expected version bumps for that project given historical data.

**Comment by greysteil** [comment]
> given a project's dependencies, what is the rate of expected version bumps for that project given historical data.

I want to comment on this properly tomorrow (it's late in the UK right now), but I really like the idea of the above!

**Comment by JacobEvelyn** [comment]
@greysteil any further thoughts?

**Comment by mizhi** [comment]
@greysteil I'm curious on your thoughts on this issue. One area where this would be very useful is in capacity planning across an engineering organization or justifying a certain cadence of maintenance work. 

As an engineering manager, I can say, "Look, we're >500 dependencies out of date across all of our repositories. Dependabot tells us that we can expect roughly 10 new PRs per week, therefore at our current cadence of 5 PRs per week, not only will we never catch up, but we will continue to fall behind." Having data like this going into a roadmapping meeting or product planning meeting would really help justifying the engineering hours to both technical and non-technical audiences.

This is, of course, coming from a very selfish interest on my part! 😆 

Also, we love dependabot. It's had a huge impact on how we pay attention to dependency updates — going from a reactive mindset to a more proactive mindset.

**Comment by greysteil** [comment]
Sorry for the slow reply on this one - got swamped last week.

I'm still really keen on this. We've been working on adding support for config files for the last couple of weeks, which has taken up most of my time (and show ship next week). Next on the list are looking at PR grouping and then expanding the dashboard so that it has a proper show view for a repo. 

Once we've got that show view I'm super keen to add statistics like the above, and perhaps a marketing page that allows folks to drop in a dependency file and see what's up-to-date, out-of-date, and how many updates they should expect each week.

So TL;DR is I'm super keen but have been crunched. We'll definitely get to this - it's a great idea.

**Comment by JacobEvelyn** [comment]
Hey @greysteil—any updates on this?

**Comment by greysteil** [comment]
Sadly not (and sorry for the slow reply here). We're doubling down on robustness work for the next month, since Dependabot has more than doubled in size over the last 3 months and I want to get our error rate back down to zero, but I'm still keen to work on this once that's done.
